src/uct/ib/base/ib_device.c

```diff
@@ -557,17 +563,17 @@ uct_ib_device_query_gid_info(struct ibv_context *ctx, const char *dev_name,
     if (ret == 0) {
         info->gid                  = attr.gid;
         info->gid_index            = gid_index;
-        info->roce_ver.addr_family =
+        info->roce_info.addr_family =
                         uct_ib_device_get_addr_family(&info->gid, gid_index);
         switch (attr.type) {
         case IBV_EXP_IB_ROCE_V1_GID_TYPE:
-            info->roce_ver.ver = UCT_IB_DEVICE_ROCE_V1;
+            info->roce_info.ver = UCT_IB_DEVICE_ROCE_V1;
```

Comment:
this function fills only the "roce_info" portion of gid_info 
so it should be
```
uct_ib_device_query_gid_roce_info(struct ibv_context *ctx, const char *dev_name,
                                  uint8_t port_num, unsigned gid_index,
                                  uct_ib_roce_info_t *roce_info)
```

Reply:
uct_ib_device_query_gid_info fills the gid and gid index as well (before the switch and with ibv_query_gid below)

Reply:
i think it doesn't have to fill gid_index, because it's already passed as an input parameter?

Reply:
yes but for completeness, so that this function would set the whole gid_info struct.

Reply:
that's why i suggest to fill only uct_ib_roce_info_t by this function

Reply:
what about the gid? it is filled in this function as well.

Reply:
sorry missed it, we can either return it separately or keep function as-is